engine/title.asm

```diff
@@ -3,17 +3,654 @@ INCLUDE "constants.asm"
+	ld [hl], a
+	
+	call .load_position_table
+		
+	
+	ld a, $1A
+	ldh [rBGP], a
+	ld a, $E4
+	ldh [rOBP0], a
+	ret
+	
+.load_position_table:
+	ld hl, FirePositionTable
+	ld c, $06	;load 6 flying objects on the screen
+	
+.set_fire_note_loop
+	push bc
+	ld e, [hl]
+	inc hl
+	ld d, [hl]
+	inc hl
+	push hl
+	ld a, $2E	;title fire/note object effect type?
+	call AddEffectObject ;3CA8 add additional object?
+	pop hl
+	pop bc
+	dec c
+	jr nz, .set_fire_note_loop
+	ret
+
+FirePositionTable:: ;5EAC-5EB7
+dw $4CE0
+dw $58A0
+dw $6490
+dw $70D0
+dw $7CB0
+dw $8800
+
+TitleFireGFX::
+INCBIN "gfx/title/fire.2bpp" ;5EB8-5F37
+TitleNotesGFX::
+INCBIN "gfx/title/notes.2bpp" ;5F38=5FB7
+
+TitleScreenMain:: ;5fb8
+	ld a, [wSequence]	; wSequence = $CB5E
+	bit 7, a
+	jr nz, .exit
+	call TitleScreenSequence
+	ld a, $23
+	ld hl, EffectObjectJumpNoDelay	; obj jump no wait? 8CD13
+	call FarCall_hl
+	call DelayFrame
+	and a
+	ret
+
+.exit
+	scf
+	ret
+
+TitleScreenSequence::
+	ld e, a
+	ld d, $00
+	ld hl, TitleScreenSequenceTable
+	add hl, de
+	add hl, de
+	ldi a, [hl]
+	ld h, [hl]
+	ld l, a
+	jp hl
+	
+TitleScreenSequenceTable::
+dw Start 
+dw LoadPokemonLogo 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw MoveTitle 
+dw MoveTitleEnd 
+dw InitFlashTitle 
+dw FlashTitle
+
+dw PMJapaneseChara 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw WaitForNextSequence 
+dw PMSubtitle 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber
+
+dw IncreaseSequenceNumber 
+dw WaitForNextSequence 
+dw Version 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw WaitForNextSequence 
+dw CopyRight
+
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw WaitForNextSequence 
+dw HoOh 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber 
+dw IncreaseSequenceNumber
+
+dw WaitForNextSequence 
+dw PressButtonInit 
+dw TitleScreenInputAndTimeout 
+dw FadeMusicOut
+
+IncreaseSequenceNumber:: ;6025
+	ld hl, wSequence	; wSequence = $CB5E
+	inc [hl]
+	ret
+
+WaitForNextSequence:: ;602A
+	xor a
+	ldh [$FFDE], a
+	ld hl, wSequence+2
+	ld a, [hl]
+	and a
+	jr z, .next_seq
+	dec [hl]
+	ret
+
+.next_seq
+	call IncreaseSequenceNumber
+	ret
+
+LoadPokemonLogo:: ;603a
+	call PrintPokemonLogo ;6196
+	call IncreaseSequenceNumber ;6025
+	ld a, $01
+	ldh [$FFDE], a
+	ret	
+	
+Start:: ;6045
+	call IncreaseSequenceNumber
+	push de
+	ld de, $002D
+	call PlaySFX	; Play "Swish" sound
+	pop de
+	ld a, $80
+	ld [wSequence+2], a
+	call SetLYOverrides
+	ld a, $43
+	ldh [$FFD0], a
+	ret
+
+MoveTitle:: ;605d
+	xor a
+	ldh [$FFDE], a
+	ld hl, wSequence+2
+	ld a, [hl]
+	and a
+	jr z, .movetitle_nextseq
+	add $04
+	ld [hl], a
+	ld e, a
+.movetitle_wait
+	ldh a, [rLY]
+	cp $40
+	jr c, .movetitle_wait
+	ld a, e
+	call SetLYOverrides
+	ret
+
+.movetitle_nextseq
+	call IncreaseSequenceNumber
+	ret	
+	
+MoveTitleEnd:: ;607a
+	xor a
+	ldh [$FFD0], a
+	call IncreaseSequenceNumber
+	ld de, $0001
```

Comment:
`MUSIC_TITLE` in [pokecrystal/constants/music_constants.asm](https://github.com/pret/pokecrystal/blob/bf074f359886707d2ffb2e0b525b03dc7af0fe29/constants/music_constants.asm#L6). (i.e. Start transferring over music constants whenever you find them)

Reply:
Sorry, I dumped all the songs yesterday and I was so worried about getting the music command macros that I forgot the song ID constants. I'll add them now